runtime/lua/vim/lsp.lua

```diff
@@ -6,6 +6,7 @@ local protocol = require('vim.lsp.protocol')
 local util = require('vim.lsp.util')
 local sync = require('vim.lsp.sync')
 local semantic_tokens = require('vim.lsp.semantic_tokens')
+local dynamic = require('vim.lsp.dynamic')
```

Comment:
I assume we want people to go through client.dynamic_capabilities, and not access vim.lsp.dynamic directly, so could you rename `dynamic.lua` to `_dynamic` and move the require call to `dynamic_capabilities = require("vim.lsp._dynamic").new()` to make it lazy?